aii-core/src/main/perl/Shellfe.pm

```diff
@@ -859,6 +885,31 @@ sub check_protected {
     return %hash;
 }
 
+sub change_dhcp 
+{
+    my ($self, $method, %nodes) = @_;
+    $self->debug(5,"logfile:", $self->option('logfile'), " dhcpcfg:", $self->option('dhcpcfg') );
+    my $dhcpmgr = AII::DHCP->new('script',
+        "--logfile=".$self->option('logfile'), 
+        "--cfgfile=".$self->option('dhcpcfg'), 
+        log => $self);
+    foreach my $node (sort keys %nodes) {
+        my $st = $nodes{$node}; 
+        $self->debug(3,"Checking dhcp config on node $node for method $method.");
+        if ($st->{configuration}->elementExists(DHCPPATH)) {
+            $self->dhcp($node, $st, $method, $dhcpmgr);
+        }
+    }    
+    if ($dhcpmgr->nodes_to_change()) {
```

Comment:
why reinvent the `AII::DHCP->configure` method? or at least fatcor out the common bits